src/target/stm32f1.c

```diff
@@ -448,16 +453,24 @@ static bool stm32f1_option_write(target *t, uint32_t addr, uint16_t value)
 	}
 	if (opt_val[index] == value)
 		return true;
+
 	/* Check for erased value */
 	if (opt_val[index] != 0xffff)
 		if (!(stm32f1_option_erase(t)))
 			return false;
 	opt_val[index] = value;
+
 	/* Write changed values*/
+	bool width_word = false;
+	if (strcmp(t->driver, "GD32E230") == 0) {
```

Comment:
This is covered by the note below about strcmp() vs testing t->idcode or similar